dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -330,21 +343,40 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
                 // Register a sink with platform which is initialized in the current process
                 auto appUserModelId{ PushNotificationHelpers::GetAppUserModelId() };
-                THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appUserModelId.get(), PushNotificationManager::Default().as<ABI::Microsoft::Internal::PushNotifications::INotificationListener>().get()));
+                bool registeredEvent{ false };
+                {
+                    auto lock{ m_lock.lock_shared() };
+                    registeredEvent = (bool)m_foregroundHandlers;
+                }
+
+                if (registeredEvent)
+                {
+                    {
+                        auto lock{ m_lock.lock_exclusive() };
+                        m_sinkRegisteredWithPlatform = true;
+                    }
+                    THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appUserModelId.get(), PushNotificationManager::Default().as<PushNotificationsABI::INotificationListener>().get()));
+                }
+
                 auto scopeExitToCleanSink{ wil::scope_exit([&]()
                 {
+                    {
+                        auto lock{ m_lock.lock_exclusive() };
+                        m_sinkRegisteredWithPlatform = false;
+                    }
                     THROW_IF_FAILED(PushNotifications_UnregisterNotificationSinkForFullTrustApplication(appUserModelId.get()));
```

Comment:
Only Unregister if this is set to true.

Reply:
Ideally the way this is structure, it shouldnt matter, bt still better for correctness sake IMHO